daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/iterable/TypeIterable.scala

```diff
@@ -255,13 +255,27 @@ private[validation] object TypeIterable {
         iterator(value)
     }
 
+  private[validation] def iterator(coImpl: InterfaceCoImplements): Iterator[Type] =
+    coImpl match {
+      case InterfaceCoImplements(template, methods) =>
+        Iterator(TTyCon(template)) ++
+          methods.values.flatMap(iterator)
```

Comment:
```suggestion 
          methods.values.iterator.flatMap(iterator)
```
Without the iterator, the type of  `methods.values.flatMap(iterator)` produces a `List` which is pretty terrible in most of the use cases `TypeIterable` which are recursive. Here we really want to to the concatenation lazily, so we need to insert a `.view` (which is pure) or a `.iterator` between the `values` and the `flatMap`.  In this case, I would favor `.iterator` as the return type is also an `iterator`